src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts

```diff
@@ -119,22 +118,34 @@ export class ConfigurationManager implements IConfigurationManager {
 	}
 
 	async resolveConfigurationByProviders(folderUri: uri | undefined, type: string | undefined, config: IConfig, token: CancellationToken): Promise<IConfig | null | undefined> {
+		// activate debuggers early for the provided type in case any '*' typed debug configuration providers are activated
+		// based on the provided type.
 		await this.adapterManager.activateDebuggers('onDebugResolve', type);
```

Comment:
This seems unnecessary since the first `resolveDebugConfigurationForType` will activate the type

Reply:
Is this just so you do the filter once below instead of inside `resolveDebugConfigurationForType`? I would probably keep it as it was before. _Most_ debug calls won't hit this more than once, and even for the ones that do an `await` costs much more than `filter`ing the relatively small list of configProvider